src/messages/MessageBuilder.cpp

```diff
@@ -2959,22 +2987,32 @@ void MessageBuilder::appendTwitchBadges(const QVariantMap &tags,
     {
         const QString sourceId = tags["source-room-id"].toString();
         QString sourceName;
+        QString sourceProfilePicture;
+        QString sourceLogin;
+
         if (sourceId.isEmpty())
         {
             sourceName = "";
         }
         else if (twitchChannel->roomId() == sourceId)
         {
+            // We have the source channel open, but we still need to load the profile picture URL
+            auto twitchUser = getApp()->getTwitchUsers()->resolveID({sourceId});
             sourceName = twitchChannel->getName();
+            sourceProfilePicture = twitchUser->profilePictureUrl;
+            sourceLogin = twitchUser->name;
         }
         else
         {
-            sourceName =
-                getApp()->getTwitchUsers()->resolveID({sourceId})->displayName;
+            auto twitchUser = getApp()->getTwitchUsers()->resolveID({sourceId});
+            sourceName = twitchUser->displayName;
+            sourceProfilePicture = twitchUser->profilePictureUrl;
+            sourceLogin = twitchUser->name;
         }
```

Comment:
The last two branches are the same except for the name and the first one isn't necessary, because `QString`'s default constructor creates an empty string. Maybe something like this:

```cpp
if (!sourceId.isEmpty())
{
    auto twitchUser = getApp()->getTwitchUsers()->resolveID({sourceId});
    sourceProfilePicture = twitchUser->profilePictureUrl;
    sourceLogin = twitchUser->name;
    if (twitchChannel->roomId() == sourceId)
    {
        sourceName = twitchChannel->getName();
    }
    else
    {
        sourceName = twitchUser->getName();
    }
}
```